### ISSUE TITLE
file import: Keep the impor file name as an attribute

### ISSUE THREAD

**Issue by urapadmin** [arch-kiosk/arch-kiosk-office#2274]
Sequence import suggests that it makes sense to keep the original filename of the file as a reminder or an information to clarify later doubts about the assignment of images to contexts.

**Comment by luizaogs** [comment]
We ran into wanting this when importing images from the SD card, yes. Keeping the original file name somewhere seems helpful, if it is possible.

**Comment by lbestock** [comment]
I would want this before going into the field.

**Comment by urapadmin** [comment]
too disruptive, postponed but could be done for post-season. Many projects do their file import after the season anyway.

**Comment by urapadmin** [comment]
- this got new priority. Needed for #3339

**Comment by urapadmin** [comment]
### Kiosk 1.7.26
- this is implemented. The original filename is shown in the file viewer's data section all the way down in the lower right corner. 
- since the data is not available for images that were imported before 1.7.26, the attribute will not show up at all
- for newly imported images it should show up
- the filename should be assigned by all import methods, even the upload (bunch). I do expect it to be set even if images come from iPads. In that case the filename will be rather useless, of course.
- you can also search for this filename by using the description in the file repository.

**Comment by luizaogs** [comment]
This isn't out yet, right? I'd be happy to test it if you release it soon - needing a (productive) distraction as Rio goes to war against itself. Again.

**Comment by urapadmin** [comment]
it is not, no. And I can't roll it out as Kiosk is a bit unwell currently because of something else I am working on 👀

**Comment by luizaogs** [comment]
Alas.

**Comment by luizaogs** [comment]
Which import methods do you want tested? I'll do one bunch upload and will upload something from an iPad too, to check that.

**Comment by urapadmin** [comment]
> Which import methods do you want tested? I'll do one bunch upload and will upload something from an iPad too, to check that.

yes, and the third you can do is upload an image to replace the existing one through the upload button. I wouldn't be surprised if that gets a cryptic temporary filename instead of the one you uploaded.

**Comment by luizaogs** [comment]
When replacing an existing image with the upload button, initially nothing shows up for the original filename, but once I refresh FM it shows up properly and I can also search for it through the description.

This is fine, I assume?

**Comment by luizaogs** [comment]
Bunch import also worked.

**Comment by luizaogs** [comment]
I'll test the iPad one soonish - iPad is currently far away and there is a cat on my lap.

**Comment by urapadmin** [comment]
> When replacing an existing image with the upload button, initially nothing shows up for the original filename, but once I refresh FM it shows up properly and I can also search for it through the description.
> 
> This is fine, I assume?

ja, refreshing the data automatically in there is hard and since this is a very rare use case I'd rather live with it than spend hours on fixing it.

**Comment by urapadmin** [comment]
🐈 lies behind me to not miss any of my movements that could lead to a snack.

**Comment by luizaogs** [comment]
Smart Bob. Mine, not so smart.

**Comment by urapadmin** [comment]
I used the orange cat emoji accurately :o) The orange one is currently the sneakier of the two.

**Comment by luizaogs** [comment]
Ha! Smart Huxley, then. I assumed it was Bob since he is attached to you usually :) 🐈‍⬛

**Comment by luizaogs** [comment]
Adding a photo from the iPad works, too - weird original file name is there.
